src/Model.php

```diff
@@ -7,14 +7,12 @@
 use Psr\Log\LoggerInterface;
 
 /**
- * Class BaseModel
- *
  * The Model classes inheriting from this abstract class are meant to represent a single database record.
  * The associated table name has to be provided in the child class, and the table is expected to have a unique `id` field.
  *
  * @property int id
  */
-abstract class BaseModel
+abstract class Model
```

Comment:
The main difference to my approach is that I try separating the dependencies from the entities. But maybe I'm wrong .. In case of "lazy-loading", we won't have a chance otherwise I now think...

Reply:
Indeed, this was my next question: how do you do lazy loading without having at least a DB dependency?

Reply:
Yep, you're right!